test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -161,13 +161,32 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 					util.PrintCommand(cmd)
 					out, err := cmd.CombinedOutput()
 					log.Printf("%s\n", out)
-					Expect(err).To(HaveOccurred())
+					Expect(err).NotTo(HaveOccurred())
 				}
 			} else {
 				Skip("root password validation only works on ubuntu distro until this lands in a VHD")
 			}
 		})
 
+		It("should validate that every linux node has the right sshd config", func() {
+			kubeConfig, err := GetConfig()
+			Expect(err).NotTo(HaveOccurred())
+			master := fmt.Sprintf("azureuser@%s", kubeConfig.GetServerName())
+			nodeList, err := node.Get()
+			Expect(err).NotTo(HaveOccurred())
+			sshdConfigs := []string{"ClientAliveInterval 120", "ClientAliveCountMax 3", "UsePrivilegeSeparation yes", "KeyRegenerationInterval 3600", "ServerKeyBits 1024", "SyslogFacility AUTH", "LogLevel INFO", "LoginGraceTime 60", "PermitRootLogin no", "PermitUserEnvironment no", "StrictModes yes", "RSAAuthentication yes", "PubkeyAuthentication yes", "IgnoreRhosts yes", "RhostsRSAAuthentication no", "HostbasedAuthentication no", "X11Forwarding no", "MaxAuthTries 4", "Banner /etc/issue.net", "AcceptEnv LANG LC_*", "Subsystem sftp /usr/lib/openssh/sftp-server", "UsePAM yes", "UseDNS no", "GSSAPIAuthentication no"}
```

Comment:
is this overkill or what you meant in your comment @jackfrancis ?

Reply:
totally not overkill :)

Reply:
Let's wrap these tests in a `if eng.ExpandedDefinition.Properties.IsUbuntuDistroForAllNodes() { } else { skip("message") }` block

Reply:
why? the sshd config is delivered via cloud init, not VHD

Reply:
I think these changes will require either a reboot or a systemctl restart sshd statement, the latter which we only want to do in the full install scenario.

Reply:
So does that mean I need to add that restart in code? Currently the e2e is failing...

Reply:
You could add a `systemctl_restart 20 5 10 restart ssh` in the `else` clause of `if $REBOOTREQUIRED; then` at the conclusion of CSE.

Reply:
added but not sure where the 4th arg "restart" is coming from. The function uses `retries=$1; wait_sleep=$2; timeout=$3 svcname=$4`

Reply:
sorry `restart` token is bogon

Reply:
looks like that didn't work